File: swift day3/代码/07-block的回顾/07-block的回顾/ViewController.m
//
//  ViewController.m
//  07-block的回顾
//
//  Created by 1 on 16/6/17.
//  Copyright © 2016年 小码哥. All rights reserved.
//

#import "ViewController.h"
#import "HttpTools.h"

@interface ViewController ()

@property (nonatomic, strong) HttpTools *httpTools;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.httpTools = [[HttpTools alloc] init];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    __weak ViewController *weakSelf = self;
    
    [self.httpTools loadData:^(NSString *result) {
        NSLog(@"在控制器中拿到数据:%@", result);
        
        // 问题: 这里是否形成了循环引用
        weakSelf.view.backgroundColor = [UIColor redColor];
    }];
}

- (void)dealloc {
    NSLog(@"ViewController --- dealloc");
}

@end

File: swift day3- 语法高级/代码/08-Swift中闭包的使用/08-Swift中闭包的使用/ViewController.swift
//
//  ViewController.swift
//  08-Swift中闭包的使用
//
//  Created by 1 on 16/6/17.
//  Copyright © 2016年 小码哥. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var httpTools : HttpTools?
    
    // override: 重写, 在swift中如果是对父类的方法进行的重写.必须在方法前加上override
    override func viewDidLoad() {
        super.viewDidLoad()
        
        httpTools = HttpTools()
    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        httpTools?.loadData({ (result) -> () in
            print("在控制器中拿到数据:\(result)")
        })
      
    }
}

